collections/_posts/2024-12-02-SSL_comparison.md

```diff
@@ -0,0 +1,123 @@
+
+| Method          | $$\lambda^L$$ | $$\lambda^U$$ |
+|-----------------|---------------|---------------|
+| Supervised      | 1             | 0             |
+| Semi-supervised | >0            | >0            |
+| Self-supervised | 0 then 1      | 1 then 0      |
+
+## Compared methods
+
+**Supervised learning**
+
+Only using the labeled $$(L)$$ dataset.
+
+* Sup: to denote a classifier trained with classical multiclass cross entropy loss.
+* MixUp: also multiclass cross entropy, but with mixup data augmentation (creating new training samples by mixing two samples from the original dataset).
+* SupCon: using a supervised contrastive learning loss (pull together samples belonging to the same class in the embedded space while pushing apart from samples of other classes).
+
+**Semi-supervised learning**
+
+They selected 6 methods of the recent years to cover a wide spectrum of strategies and cost, to take advantage of the unlabeled $$(U)$$ and labeled $$(L)$$ datasets jointly.
+
+* Pseudo-label: sets the label to unlabeled data to the class which has the maximum predicted probability.
+* Mean teacher: teacher weights are updated as exponential moving average of student weights after each sample, either the sample used is labeled or not.
+* MixMatch: uses mixup data augmentation to combine label and unlabeled data to compute labels for this new input images.
+* FixMatch: combination of pseudo-label method to set a label to unlabeled images if the prediction confidence is high enough with strong data augmentation, which prediction has to match the label set previously.
+* FlexMatch: adding a curriculum notion to FixMatch to consider different learning status and learning difficulties of different classes, basically adjusting the threshold of confidence to define a class to an unlabeled image.
+* CoMatch: learns jointly class probability and low-dimensional embeddings to impose a smoothness constraint and improve the pseudo-labels, while those last  regularize the structure of the embeddings through graph based contrastive learning.
+
+**Self-supervised learning**
+
+In addition, 7 self-supervised methods are compared:
+
+* SimCLR: contrastive learning self-supervised method based on data augmentation with a learning nonlinear transformation between the representation and contrastive loss to improve the learned representations.
+* MOCO v2: for momentum contrastive, which uses a dynamic dictionary with a queue and a moving-averaged encoder that facilitates contrastive unsupervised learning.
+* SwAV: contrastive learning that does not compare features directly but comparing different augmentation of the same images.
```

Comment:
"...but comparing different augmentation of the same images." → Correct: "...but compares different augmentations of the same image."